UT-IfcPavement / UT-IfcCourse

BernhardWehrle opened the pull request:
## Unit test

*provide a short description*

## Covered topics

*tick one/many*

- [ ] project setup
- [ ] georeferencing
- [ ] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [ ] project breakdown structure / spatial structure
- [ ] terrain
- [ ] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [x] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [ ] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

BernhardWehrle committed:
UT-IfcPavement / UT-IfcCourse

jmirtsch commented on UT-IfcCourse/UT-IfcCourse.ifc:
```diff
@@ -0,0 +1,56 @@
+#35 = IFCCOURSE('1Y6olD18bsgajKcDf8M1yx', $, 'Querprofile Achse 1', 'Querprofile Achse 1', $, #16, #36, $, $);
+#36 = IFCPRODUCTDEFINITIONSHAPE($, $, (#37));
+#37 = IFCSHAPEREPRESENTATION(#38, 'Body', 'Curve2D', (#39));
+#38 = IFCREPRESENTATIONCONTEXT(IFCLABEL('Curve2D'), IFCLABEL('Curve2D'));
```
IfcRepresentationContext is an abstract class, and only subtypes can be instantiated.  ie IfcGeometricRepresentationContext

SergejMuhic commented on UT-IfcCourse/UT-IfcCourse.ifc:
```diff
@@ -0,0 +1,56 @@
+#17 = IFCAXIS2PLACEMENT3D(#18, $, $);
+#18 = IFCCARTESIANPOINT((51.888, -7.639, 0.));
+#19 = IFCPROJECT('1W7lFkDRwMWKYdP6AbIlsD', #5, 'Projekt: UT-IfcCourse', 'Projekt: UT-IfcCourse', $, $, $, (#13), #7);
+#20 = IFCSITE('1d1nf$UMdHWK5jSGoE5PmO', #5, 'VESTRA', 'VESTRA', $, #16, $, $, $, $, $, $, IFCLABEL('VESTRA'), $);
```
Selects or enums require a type before the value. Otherwise, the simple defined type is not serialized in the file. For IfcLabel that means the following:
IFCSITE('1d1nf$UMdHWK5jSGoE5PmO', #5, 'VESTRA', 'VESTRA', $, #16, $, $, $, $, $, $, 'VESTRA', $);

SergejMuhic reviewed (changes requested):
Simple defined types shall not have types serialized.

pjanck commented:
Following the changes to the schema, (we are now at `RC3`), we ask you to update your IFC files accordingly. The schema can be found [here](https://github.com/bSI-InfraRoom/IFC-Specification/blob/main/IFC4x3_RC3.exp). The checking procedure for IFC files has been updated - please merge in the current state of the `main` branch.

BernhardWehrle committed:
Update UT-IfcCourse and IfcPavement to IFC4X3RC3

BernhardWehrle committed:
Directrix changed from IfcCompositeCurve to IfcGradientCurve

AlexBrad1eyCT reviewed (commented):
All Semantically correct, alignment experts need to review the alignment and the errors you mentioned

BernhardWehrle committed:
Implementet new IFC4_RC4. Removed IfcAlignmentHorizontal.StartDistanceAlong

pjanck reviewed (changes requested):
I believe that `IfcCourse` and `IfcPavement` got removed from the files together with `IfcAlignmentVertical` - is that an intentional result?

BernhardWehrle commented:
I updated the schema and removed the StartDistAlgong property from the IfcAlignment entity. Did I misunderstand something in the last email? I read in the IFC specification that the property was removed.

You also can call me on 07634/5612 816

pjanck commented on UT-IfcCourse/UT-IfcCourse.ifc:
```diff
@@ -53,26 +53,5 @@ DATA;
-#66 = IFCCARTESIANPOINTLIST2D(((0., -1.), (-3., -1.1), (-3., -1.E-1), (0., 0.), (3., 1.E-1), (3., -9.E-1), (0., -1.)), (IFCLABEL('1'), IFCLABEL('2'), IFCLABEL('3'), IFCLABEL('4'), IFCLABEL('5'), IFCLABEL('6'), IFCLABEL('7')));
-#67 = IFCARBITRARYCLOSEDPROFILEDEF(.AREA., IFCLABEL('ClosedProfile'), #68);
-#68 = IFCINDEXEDPOLYCURVE(#69, $, $);
-#69 = IFCCARTESIANPOINTLIST2D(((0., -1.), (-3., -1.1), (-3., -1.E-1), (0., 0.), (3., 1.E-1), (3., -9.E-1), (0., -1.)), (IFCLABEL('1'), IFCLABEL('2'), IFCLABEL('3'), IFCLABEL('4'), IFCLABEL('5'), IFCLABEL('6'), IFCLABEL('7')));
```
These lines got removed in the last commit. Is this intentional?

pjanck commented on UT-IfcPavement/UT-IfcPavement.ifc:
```diff
@@ -53,26 +53,5 @@ DATA;
-#66 = IFCPOINTBYDISTANCEEXPRESSION(IFCLENGTHMEASURE(0.), IFCLENGTHMEASURE(0.), IFCLENGTHMEASURE(91.), $, #35);
-#67 = IFCOPENCROSSPROFILEDEF(.CURVE., $, .T., (-3., 3.), (-3.006, -3.006), (IFCLABEL('-1.AL'), IFCLABEL('AX'), IFCLABEL('+1.AR')));
-#68 = IFCPOINTBYDISTANCEEXPRESSION(IFCLENGTHMEASURE(14.142), IFCLENGTHMEASURE(0.), IFCLENGTHMEASURE(92.), $, #35);
-#69 = IFCOPENCROSSPROFILEDEF(.CURVE., $, .T., (-3., 3.), (-3.006, -3.006), (IFCLABEL('-1.AL'), IFCLABEL('AX'), IFCLABEL('+1.AR')));
```
These as well.

pjanck reviewed (changes requested):
I'm referring to these changes pushed in the last commit 81baaf6

BernhardWehrle commented:
Oh shit. Something went wrong. I will create new files. Sorry.

BernhardWehrle committed:
Fixed bug in unit test files

larswik commented:
Misses a parameter for IfcAlignment (has 7 should be 8). Also get two of these from KIT checker which I didn't analyze further at this point: "access to attribute 'offsetlongitudinal' of non-entity instance 'temp'"

BernhardWehrle commented:
Hi all

i think in the latest pullrequest the IfcAlignment should have 8 parameters


[cid:66bc23a8-b0d9-4cd8-94f3-89034cd0cad3]










AKG Software Consulting GmbH
Bernhard Wehrle
Entwicklung

Uhlandstraße 12
79423 Heitersheim
Germany

Tel.: +49 (0)7634 5612-0
Fax: +49 (0)7634 5612-300
E-Mail: ***@***.***
Internet: www.akgsoftware.de<http://www.akgsoftware.de>

Geschäftsführer: Dipl.-Ing. Artur K. Günther, Dipl.-Ing. Arno Brüggemann, Dipl.-Ing. (FH) Bernhard Feser, Dipl.-Ing. Jens Günther
Sitz der Gesellschaft: Heitersheim, HRB 310383 (Amtsgericht Freiburg)


________________________________
Von: larswik ***@***.***>
Gesendet: Montag, 31. Mai 2021 10:58
An: bSI-InfraRoom/IFC-infra-unit-test ***@***.***>
Cc: Bernhard Wehrle ***@***.***>; Author ***@***.***>
Betreff: Re: [bSI-InfraRoom/IFC-infra-unit-test] UT-IfcPavement / UT-IfcCourse (#52)



Externe E-Mail: Vorsicht beim Öffnen von Anhängen und Klicken auf Links!



Misses a parameter for IfcAlignment (has 7 should be 8). Also get two of these from KIT checker which I didn't analyze further at this point: "access to attribute 'offsetlongitudinal' of non-entity instance 'temp'"

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/bSI-InfraRoom/IFC-infra-unit-test/pull/52#issuecomment-851337328>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJHOJH7UZRXFKN24IDJWGU3TQNFVBANCNFSM4XYJYNBA>.

BernhardWehrle committed:
add StartDistAlong to IfcAlignmentHorizontal
